### PR TITLE
Introduce CCN profiles for RHEL9

### PR DESCRIPTION
**Pull request by marcusburghardt**
#### Description:

The https://github.com/ComplianceAsCode/content/pull/10801 introduced a new control file for CCN Policy for RHEL9.
This PRs reviewed all `related_rules` in the planned items, included the CCN reference in these rules, moved them to `rules` and updated the requirement `status`.

Finally, it was created three new profiles for RHEL9, each based on a CCN level:
- Basic
- Interemediate
- Advanced

At this initial version, about 70% of the requirements are automated.
```
$ utils/controleval.py stats -p rhel9 -i ccn_rhel9 -l advanced
General stats:
all                  81 /  81 = 100.0%
applicable           81 /  81 = 100.0%
assessed             61 /  81 = 75.31%
full coverage        57 /  81 = 70.37%
not assessed         20 /  81 = 24.69%
```

#### Rationale:

New profile to attend RHEL9 users interested in `CCN-STIC-610A22`

#### Review Hints:

I carefully ensured that only rules already enabled for RHEL9 and used in other common profiles, like CIS and STIG, were included in this initial version. That way we can have stable profiles much less prone to unexpected issues.

I recommend to review this PR in two steps.
First we can check if the profiles are working as expected:

1. ./build_product rhel9
2. ./tests/ds_unselect_rules.sh build/ssg-rhel9-ds.xml tests/unselect_rules_list
2.1. This is to disable some rules which cause issues in the test environment.
3. Test the three profiles in a `rhel9` VM: 
3.1. ./tests/automatus.py profile --libvirt qemu:///session rhel9 --datastream build/ssg-rhel9-ds.xml --remediate-using bash ccn_advanced
3.2. ./tests/automatus.py profile --libvirt qemu:///session rhel9 --datastream build/ssg-rhel9-ds.xml --remediate-using bash ccn_intermediate
3.3. ./tests/automatus.py profile --libvirt qemu:///session rhel9 --datastream build/ssg-rhel9-ds.xml --remediate-using bash ccn_basic

The second step would be to check the references included in the rules.
For this step the best approach is to check commit by commit, individually.

Don't be scared by the number of commits. They are quite small and are separated for better organization, with a better review experience in mind. ;)

### CODE REVIEW

**Comment by marcusburghardt** [comment]
`codeclimate` is complaining about a `Line too long (183 > 99 characters) ` but the referenced line includes the URL for the CCN Policy, which is indeed very long but correct. So, this `codeclimate` hit can be safely waived in this case.

**Comment by marcusburghardt** [comment]
I will take a look in these product stability tests and try restrict the changes to rhel9 only.

**Comment by marcusburghardt** [comment]
> I will take a look in these product stability tests and try restrict the changes to rhel9 only.

Done

File: ssg/constants.py [hunk 1]
```diff
@@ -242,6 +242,9 @@
     anssi=Reference(
         id="anssi", name="ANSSI", url=anssi_ns,
         regex_with_groups=r"BP28\(R(\d+)\)"),
+    ccn=Reference(
+        id="ccn", name="CCN", url="",
+        regex_with_groups=r"A.(\d+).SEC-RHEL(\d+)"),
```
Comment (Mab879) [inline ssg/constants.py:247]: This fine for now, but if other products (SUSE, etc) need a CCN reference we will need to change this.

**Review by Mab879 — APPROVED** [review]

**Pull request merged**